### PR TITLE
UI-2391 Extend px-map-marker-group component to support numbered marker

### PR DESCRIPTION
**Pull request by mallik-svmx**
- Re-defined structure to support rendering numbered marker:
`"marker-icon": {
  "icon-base": "custom-icon",
  "icon-type": "px-nav:unconfirmed", //also component defaults to px-nav:unconfirmed
  "icon-strokeColor": "blue",
  "icon-fillColor": "white",
  "icon-content": "1", //"icon-content": "px-nav:home" for icon marker
  "icon-options": { //currently it doesn't work, will be taken after/or part of UI-2363
    "strokeColor": "blue",
    "fillColor": "white",
  }
}`
- Icon-options is only valid when an "icon-content": "px-nav:home" and not a number.
- Enhanced demo to illustrate the number markers with marker group.
- Updated release notes.
- Reference demo screenshots.
screen shot 2018-08-17 at 12 09 23 pm
screen shot 2018-08-17 at 1 12 45 pm

### CODE REVIEW

**Comment by mallik-svmx** [comment]
Addressed. @shao

**Comment by shao** [comment]
Looks better - it still looks a little bit off-center to the left?  If you expand the image and you can see this?

**Comment by mallik-svmx** [comment]
Good catch :-) @shao Now it's fixed using text-align center vs. margin props.

+ Added custom icon on to clustered marker and updated Jira with new marker-icon Json. @sabrinalui-ge thank you! for calling that out.

File: HISTORY.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+v3.6.1
```
[the remark this answers is not in the document]
Comment (mallik-svmx) [reply HISTORY.md]: Done

File: px-map-behavior-icon.es6.js [hunk 2]
```diff
@@ -251,4 +251,66 @@
...
+   */
+  class CustomIcon {
+    constructor(settings={}) {
+      this.icon = this.createIcon(settings);
+      return this.icon;
+    }
+
+    createIcon(settings={}) {
+      // type defaults to type='px-nav:unconfirmed'
+      let { type='px-nav:unconfirmed', styleScope, strokeColor, fillColor, content, options: { icon, strokeColor: iconStrokeColor, fillColor: iconFillColor } } = settings;
```
[the remark this answers is not in the document]
Comment (mallik-svmx) [reply px-map-behavior-icon.es6.js]: Fixed

**Comment by hanzge** [comment]
Merging with 2 approvals.

**Pull request merged**